******************************** Log Start ********************************
Timestamp: 02/02/2022 08:57:42:082
Exception Message: 'GetLstBusinessObject' method could not be resolved. Potential candidates are: 
GetLstBusinessObject(ablnExcludeAbstractTypes:Boolean):List`1
Stack Trace:    at Microsoft.AspNet.SignalR.Hubs.NullMethodDescriptor.<get_Invoker>b__0(IHub emptyHub, Object[] emptyParameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 02/03/2022 01:12:22:891
Exception Message: 'GetLstBusinessObject' method could not be resolved. Potential candidates are: 
GetLstBusinessObject(ablnExcludeAbstractTypes:Boolean):List`1
Stack Trace:    at Microsoft.AspNet.SignalR.Hubs.NullMethodDescriptor.<get_Invoker>b__0(IHub emptyHub, Object[] emptyParameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 02/17/2022 09:28:59:097
Exception Message: Invalid column name 'REFRESHABLE'.
Stack Trace:    at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.Common.DbDataAdapter.FillInternal(DataSet dataset, DataTable[] datatables, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet, Int32 startRecord, Int32 maxRecords, String srcTable, IDbCommand command, CommandBehavior behavior)
   at System.Data.Common.DbDataAdapter.Fill(DataSet dataSet)
   at Sagitec.Common.SystemSettings.InitializeSystemSettings(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.Initialize(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.get_Instance()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 12/15/2022 05:21:48:914
Exception Message: 'GetLstBusinessObject' method could not be resolved. Potential candidates are: 
GetLstBusinessObject(ablnExcludeAbstractTypes:Boolean):List`1
Stack Trace:    at Microsoft.AspNet.SignalR.Hubs.NullMethodDescriptor.<get_Invoker>b__0(IHub emptyHub, Object[] emptyParameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 02/10/2023 11:26:39:760
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/04/2023 03:54:00:182
Exception Message: Check the Scenario XML Path in settings file.
Stack Trace:    at SignalRSelfHost.Classes.UtilityFunctions.CreateScenarioPathFromRule(String astrRuleFilePath)
   at SignalRSelfHost.Hubs.hubRuleModel.CreateNewScenario(String strScenarioType, String strEntityName, String strRuleID, String strRulePath, Boolean blnIsUpdateModeForExcel)
   at lambda_method(Closure , IHub , Object[] )
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/07/2023 01:09:40:669
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 06/05/2023 09:48:32:577
Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 06/05/2023 09:49:01:980
Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.SystemSettings.InitializeSystemSettings(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.Initialize(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.get_Instance()
   at SignalRSelfHost.Classes.UtilityFunctions.GetAppSettings(String astrKey)
   at SignalRSelfHost.Classes.ClsStaticData.PopulateCustomTypesList()
   at SignalRSelfHost.Classes.ClsStaticData.LoadTypes()
   at SignalRSelfHost.Hubs.hubMain.RefreshMasterData()
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)
   at Microsoft.AspNet.SignalR.Hubs.HubMethodDispatcher.<>c__DisplayClass1.<WrapVoidAction>b__0(IHub hub, Object[] parameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 06/05/2023 09:49:27:680
Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 06/05/2023 09:49:43:379
Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.SystemSettings.InitializeSystemSettings(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.Initialize(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.get_Instance()
   at SignalRSelfHost.Classes.UtilityFunctions.GetAppSettings(String astrKey)
   at SignalRSelfHost.Classes.ClsStaticData.PopulateCustomTypesList()
   at SignalRSelfHost.Classes.ClsStaticData.LoadTypes()
   at SignalRSelfHost.Hubs.hubMain.RefreshMasterData()
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)
   at Microsoft.AspNet.SignalR.Hubs.HubMethodDispatcher.<>c__DisplayClass1.<WrapVoidAction>b__0(IHub hub, Object[] parameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 06/12/2023 04:27:05:568
Exception Message: The type initializer for 'Sagitec.DBUtility.DBFunction' threw an exception.
Inner Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at Sagitec.DBUtility.DBFunction.LoadSystemSettings()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 07/12/2023 09:16:14:090
Exception Message: The type initializer for 'Sagitec.DBUtility.DBFunction' threw an exception.
Inner Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at Sagitec.DBUtility.DBFunction.LoadSystemSettings()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 07/14/2023 09:38:02:777
Exception Message: The type initializer for 'Sagitec.DBUtility.DBFunction' threw an exception.
Inner Exception Message: Cannot open database "PSPB_Dev_032023" requested by the login. The login failed.
Login failed for user 'devuser'.
Stack Trace:    at Sagitec.DBUtility.DBFunction.LoadSystemSettings()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation)
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject)

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/24/2025 08:00:40:045
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)
Inner Exception Message: The network path was not found
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\UtilityFunctions.cs:line 945
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 76

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/24/2025 08:01:27:111
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)
Inner Exception Message: The network path was not found
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.SystemSettings.InitializeSystemSettings(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.Initialize(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.get_Instance()
   at SignalRSelfHost.Classes.UtilityFunctions.GetAppSettings(String astrKey) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\UtilityFunctions.cs:line 7594
   at SignalRSelfHost.Classes.ClsStaticData.PopulateCustomTypesList() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\ClsStaticData.cs:line 935
   at SignalRSelfHost.Classes.ClsStaticData.LoadTypes() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\ClsStaticData.cs:line 951
   at SignalRSelfHost.Hubs.hubMain.RefreshMasterData() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 4019
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 116
   at Microsoft.AspNet.SignalR.Hubs.HubMethodDispatcher.<>c__DisplayClass1.<WrapVoidAction>b__0(IHub hub, Object[] parameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/24/2025 08:04:13:194
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)
Inner Exception Message: The network path was not found
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.DatabaseSettings.InitializeDatabaseSettings()
   at Sagitec.Common.DatabaseSettings..ctor()
   at Sagitec.Common.AppSettingsHelper.Reset()
   at SignalRSelfHost.Classes.UtilityFunctions.SetAppSettingsLocation(String astrAppSettingsLocation) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\UtilityFunctions.cs:line 945
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 76

******************************** Log End ********************************

******************************** Log Start ********************************
Timestamp: 04/24/2025 08:04:55:539
Exception Message: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)
Inner Exception Message: The network path was not found
Stack Trace:    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at Sagitec.Common.SystemSettings.InitializeSystemSettings(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.Initialize(Boolean ablnRefreshableOnly)
   at Sagitec.Common.SystemSettings.get_Instance()
   at SignalRSelfHost.Classes.UtilityFunctions.GetAppSettings(String astrKey) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\UtilityFunctions.cs:line 7594
   at SignalRSelfHost.Classes.ClsStaticData.PopulateCustomTypesList() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\ClsStaticData.cs:line 935
   at SignalRSelfHost.Classes.ClsStaticData.LoadTypes() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Classes\ClsStaticData.cs:line 951
   at SignalRSelfHost.Hubs.hubMain.RefreshMasterData() in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 4019
   at SignalRSelfHost.Hubs.hubMain.LoadProjectData(clsProjectDetails objProjectDetails, Boolean reloadCurrentProject) in C:\sagitec-source\SPPFrameworkCollection\Platform6.0\Main\Tools\SWS\SignalRSelfHost\Hubs\hubMain.cs:line 116
   at Microsoft.AspNet.SignalR.Hubs.HubMethodDispatcher.<>c__DisplayClass1.<WrapVoidAction>b__0(IHub hub, Object[] parameters)
   at Microsoft.AspNet.SignalR.Hubs.HubDispatcher.Incoming(IHubIncomingInvokerContext context)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNet.SignalR.Hubs.HubPipelineModule.<>c__DisplayClass1.<<BuildIncoming>b__0>d__3.MoveNext()

******************************** Log End ********************************

